From 6c89e7aad3e6ca19cb5ed4e021ded3f7e84c907d Mon Sep 17 00:00:00 2001
From: Shawn Guo <shawn.guo@freescale.com>
Date: Mon, 2 Sep 2013 13:52:52 +0800
Subject: [PATCH 0553/1543] ENGR00269945: ARM: dts: imx6: add ocotp fuse node

commit 96a53516e7eef055c2c93dc58e395f5a58a78640 from
git://git.freescale.com/imx/linux-2.6-imx.git

The ocotp also plays as a fuse device beside the system controller we
already have in device tree.  Let's add a device node for the fuse
device for imx6qdl and imx6sl.

While at it, the patch also drops "fsl,imx6q-ocotp" and
"fsl,imx6sl-ocotp" from system controller compatible string, since
it's not used, and use node name ocotp-ctrl and ocotp-fuse for system
controller and fuse device respectively.

shawn.guo: cherry-pick commit 1a8c2aef633c from imx_3.10.y, and add the
same change for imx6sx.

Signed-off-by: Shawn Guo <shawn.guo@freescale.com>
---
 arch/arm/boot/dts/imx6qdl.dtsi |   10 ++++++++--
 arch/arm/boot/dts/imx6sl.dtsi  |   10 ++++++++--
 arch/arm/boot/dts/imx6sx.dtsi  |   10 ++++++++--
 3 files changed, 24 insertions(+), 6 deletions(-)

diff --git a/arch/arm/boot/dts/imx6qdl.dtsi b/arch/arm/boot/dts/imx6qdl.dtsi
index 6e317f0d..4efca65 100644
--- a/arch/arm/boot/dts/imx6qdl.dtsi
+++ b/arch/arm/boot/dts/imx6qdl.dtsi
@@ -961,8 +961,14 @@
 				clocks = <&clks IMX6QDL_CLK_EIM_SLOW>;
 			};
 
-			ocotp: ocotp@021bc000 {
-				compatible = "fsl,imx6q-ocotp", "syscon";
+			ocotp: ocotp-ctrl@021bc000 {
+				compatible = "syscon";
+				reg = <0x021bc000 0x4000>;
+				clocks = <&clks IMX6QDL_CLK_IIM>;
+			};
+
+			ocotp-fuse@021bc000 {
+				compatible = "fsl,imx6q-ocotp";
 				reg = <0x021bc000 0x4000>;
 				clocks = <&clks IMX6QDL_CLK_IIM>;
 			};
diff --git a/arch/arm/boot/dts/imx6sl.dtsi b/arch/arm/boot/dts/imx6sl.dtsi
index d99fa81..252d6bf 100644
--- a/arch/arm/boot/dts/imx6sl.dtsi
+++ b/arch/arm/boot/dts/imx6sl.dtsi
@@ -809,8 +809,14 @@
 				interrupts = <0 14 IRQ_TYPE_LEVEL_HIGH>;
 			};
 
-			ocotp: ocotp@021bc000 {
-				compatible = "fsl,imx6sl-ocotp", "syscon";
+			ocotp: ocotp-ctrl@021bc000 {
+				compatible = "syscon";
+				reg = <0x021bc000 0x4000>;
+				clocks = <&clks IMX6SL_CLK_OCOTP>;
+			};
+
+			ocotp-fuse@021bc000 {
+				compatible = "fsl,imx6sl-ocotp", "fsl,imx6q-ocotp";
 				reg = <0x021bc000 0x4000>;
 				clocks = <&clks IMX6SL_CLK_OCOTP>;
 			};
diff --git a/arch/arm/boot/dts/imx6sx.dtsi b/arch/arm/boot/dts/imx6sx.dtsi
index ed87ce6..5162263 100644
--- a/arch/arm/boot/dts/imx6sx.dtsi
+++ b/arch/arm/boot/dts/imx6sx.dtsi
@@ -935,8 +935,14 @@
 				clocks = <&clks IMX6SX_CLK_EIM_SLOW>;
 			};
 
-			ocotp: ocotp@021bc000 {
-				compatible = "fsl,imx6sx-ocotp", "syscon";
+			ocotp: ocotp-ctrl@021bc000 {
+				compatible = "syscon";
+				reg = <0x021bc000 0x4000>;
+				clocks = <&clks IMX6SX_CLK_OCOTP>;
+			};
+
+			ocotp-fuse@021bc000 {
+				compatible = "fsl,imx6sx-ocotp", "fsl,imx6q-ocotp";
 				reg = <0x021bc000 0x4000>;
 				clocks = <&clks IMX6SX_CLK_OCOTP>;
 			};
-- 
1.7.5.4

